spec/data-model/README.md

```diff
@@ -94,9 +94,10 @@ Each message _declaration_ is represented by a `Declaration`,
 which connects the `name` of a _variable_
 with its _expression_ `value`.
 The `name` does not include the initial `$` of the _variable_.
-
-The `name` of an `InputDeclaration` MUST be the same
-as the `name` in the `VariableRef` of its `VariableExpression` `value`.
+If the `value` has a `VariableRef` `arg` with the same `name`
+as the `Declaration`,
+it represents an _input-declaration_.
+Otherwise, it represents a _local-declaration_.
```

Comment:
I'm not fond of the blanket `value: Expression`, which requires this wording here. I'd much prefer if the data model didn't need these additional validity constraints. I think the current data model achieves this to a larger extent.

I wouldn't want anyone to think that it's OK to have `.input {:func}` or `.input {|1|}` -- which I think the proposed change does suggest.

Reply:
This change is in fact _removing_ a MUST validity constraint on `InputDeclaration`, in addition to removing a similar implicit constraint on the `LocalDeclaration` that it must not contain in its `value` a `VariableExpression` which uses the same `name` as used at the top level of the declaration.

Could you explain how this proposed language suggests that something like `.input {:func}` is ok? As it explicitly requires the expression to have a `VariableRef` argument with a matching `name`, isn't that the same set of requirements that are expressed in the syntax?